### PR TITLE
attemp to fix 5564 - consistent way to create k8s client

### PR DESCRIPTION
**Pull request by linsun**
https://github.com/istio/istio/issues/5564

### CODE REVIEW

File: pilot/pkg/config/kube/crd/client.go [hunk 1]
```diff
@@ -137,11 +138,15 @@ func (rc *restClient) createRESTConfig(kubeconfig string) (config *rest.Config,
 	if kubeconfig == "" {
 		config, err = rest.InClusterConfig()
 	} else {
-		config, err = clientcmd.BuildConfigFromFlags("", kubeconfig)
+		loadingRules := clientcmd.NewDefaultClientConfigLoadingRules()
+		loadingRules.ExplicitPath = kubeconfig
+
+		clientConfig := clientcmd.NewNonInteractiveDeferredLoadingClientConfig(loadingRules, &clientcmd.ConfigOverrides{})
+		config, err = clientConfig.ClientConfig()
```
Comment (ayj) [inline pilot/pkg/config/kube/crd/client.go]: Looks like [`BuildConfigFromFlags`](https://github.com/kubernetes/client-go/blob/b4efb2d46b33411d25a37b364098993ca896e900/tools/clientcmd/client_config.go#L545) handles all of these cases (e.g. in and out of cluster plus default client loading rules). Using that in istioctl might also fix https://github.com/istio/istio/issues/1971.

**Comment by costinm** [comment]
Looks good - I would have liked a method that also accepts KUBECONFIG ( we use it in a lot of tests
and is a common pattern ). But it doesn't seem to be a regression.

/lgtm
/approve

**Comment by linsun** [comment]
@costinm thanks for the review!  Re your comment, isn't that what createInterface() does today or are you referring to KUBECONFIG as an env var?

**Pull request merged**